library/psa_crypto.c

```diff
@@ -2380,12 +2380,21 @@ static psa_status_t psa_mac_setup( psa_mac_operation_t *operation,
         status = PSA_ERROR_BAD_STATE;
         goto exit;
     }
-
     status = psa_get_and_lock_key_slot_with_policy(
                  key,
                  &slot,
                  is_sign ? PSA_KEY_USAGE_SIGN_MESSAGE : PSA_KEY_USAGE_VERIFY_MESSAGE,
                  alg );
+    if( status == PSA_ERROR_NOT_PERMITTED )
+    {
+        /* Try with the other option (using sign/verify message) */
+        status = psa_get_and_lock_key_slot_with_policy(
+                    key,
+                    &slot,
+                    is_sign ? PSA_KEY_USAGE_SIGN_MESSAGE : PSA_KEY_USAGE_VERIFY_MESSAGE,
+                    alg );
+    }
+
```

Comment:
This is very weird.

I suspect that when this commit was written the psa_get_and_lock_key_slot_with_policy call above either didn't exist, or looked differently.

And that since then we have resolved a conflict badly.

This commit is invoking the exact same function with the exact same arguments AFAICT which is very rare to be desired behaviour.

https://github.com/nrfconnect/sdk-mbedtls/pull/15/commits/94d7e388c5c408ca45843f393bc0ad5de23a2a10

Reply:
I see the hunk below changes the first invocation from MESSAGE to HASH.

Maybe we need to do the same?

Perhaps we could ask the author what is going on here.

Reply:
Yes. This seems a bit off. 

I thin kthe sign/verify message is already tried, so we only try with the sign/verify hash afterwards